pkg/package-server/provider/inmem.go

```diff
@@ -248,6 +248,7 @@ func (m *InMemoryProvider) Get(namespace, name string) (*packagev1alpha1.Package
 
 	for key, pm := range m.manifests {
 		if key.packageName == name && (key.catalogSourceNamespace == namespace || key.catalogSourceNamespace == m.globalNamespace) {
+			pm.SetNamespace(namespace)
```

Comment:
The one thing that I think could become a problem in the future is when we have multiple packages of the same name in both the "global" and request namespace. In that situation you would have a 50% chance of getting the "global" package instead of the one you actually requested. Maybe for a get of an individual resource, you should only get the "global" package if no package of the same name exists in the requested namespace.